catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ExportCommand.java

```diff
@@ -0,0 +1,569 @@
+        } else {
+            resolvedOutput = output;
+        }
+
+        return new File(resolvedOutput);
+    }
+
+    private List<ExportItem> doMetacardExport(ZipFile zipFile, Filter filter) {
+        List<ExportItem> exportedItems = new ArrayList<>();
+        for (Result result : new QueryResulterable(catalogFramework,
+                (i) -> getQuery(filter, i, PAGE_SIZE),
+                PAGE_SIZE)) {
+            writeToZip(zipFile, result);
+            exportedItems.add(new ExportItem(result.getMetacard()
+                    .getId(),
+                    getTag(result),
+                    result.getMetacard()
+                            .getResourceURI()));
+            // Fetch and export all history for each exported item
+            for (Result revision : new QueryResulterable(catalogFramework,
+                    (i) -> getQuery(getHistoryFilter(result), i, PAGE_SIZE),
+                    PAGE_SIZE)) {
+                writeToZip(zipFile, revision);
+                exportedItems.add(new ExportItem(revision.getMetacard()
+                        .getId(),
+                        getTag(revision),
+                        revision.getMetacard()
+                                .getResourceURI()));
+            }
+        }
+        return exportedItems;
+    }
+
+    private List<ExportItem> doContentExport(/*Mutable,IO*/ZipFile zipFile,
+            List<ExportItem> exportedItems) throws ZipException {
+        List<ExportItem> contentItemsToExport = exportedItems.stream()
+                // Only things with a resource URI
+                .filter(ei -> ei.getResourceURI() != null)
+                // Only our content scheme
+                .filter(ei -> ei.getResourceURI()
+                        .getScheme()
+                        .startsWith(ContentItem.CONTENT_SCHEME))
+                // Deleted Metacards have no content associated
+                .filter(ei -> !ei.getMetacardTag()
+                        .equals("deleted"))
+                // for revision metacards, only those that have their own content
+                .filter(ei -> !ei.getMetacardTag()
+                        .equals("revision") || ei.getResourceURI()
+                        .getSchemeSpecificPart()
+                        .equals(ei.getId()))
+                .filter(distinctByKey(ei -> ei.getResourceURI()
+                        .getSchemeSpecificPart()))
+                .collect(Collectors.toList());
+
+        List<ExportItem> exportedContentItems = new ArrayList<>();
+        for (ExportItem contentItem : contentItemsToExport) {
+            ResourceResponse resource;
+            try {
+                resource =
+                        catalogFramework.getLocalResource(new ResourceRequestById(contentItem.getId()));
+            } catch (IOException | ResourceNotSupportedException e) {
+                throw new CatalogCommandRuntimeException(
+                        "Something went wrong while fetching resources",
```

Comment:
btw this exception halts the export. Though the `ResourceNotSupportedException` should probably go with the `ResourceNotFoundException` below in the catch block that continues on with the export